cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -248,10 +260,54 @@ public static void main(String[] args) {
         cliMain.tryAutoconnect();
         cliMain.updateCLIPrompt(cliConfig.getClientConfig());
 
+        if (launchOptions.getScriptFile() != null) {
+
+          ConnectionConfig connectionConfig = clientConfig.getConnectionConfig();
+          URI baseURI = connectionConfig.getURI();
+          URI uri = baseURI.resolve("/" + connectionConfig.getNamespace());
+          String cliPrompt = "\ncdap (" + uri + ")> ";
+
+          BufferedReader br = null;
+          try {
+            String line;
+            br = new BufferedReader(new FileReader(launchOptions.getScriptFile()));
+            while ((line = br.readLine()) != null) {
+              String ln = line.trim();
+              if ("".equals(ln) || ln.startsWith("#")) {
+                continue;
+              }
+              if (ln.startsWith("break")) {
+                output.println(cliPrompt + ln);
+                break; // Break from script
+              }
+              output.println(cliPrompt + ln);
+              cli.execute(ln, output);
+            }
+            output.println("\n");
+          } catch (IOException e) {
+            output.println(e.getMessage());
+            usage();
```

Comment:
Why print usage on `IOException`? This is thrown when `readLine` fails right? I think we should just print an error message like "Couldn't read from script file <path-to-script-file>"